bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MathUtils.java

```diff
@@ -63,4 +67,28 @@ public static long remainder(long numerator, long denominator) {
             }
         }
     }
+
+    public static long addExact(long num1, long num2) {
+        long result = num1 + num2;
+        if (((num1 ^ result) & (num2 ^ result)) < 0) {
```

Comment:
What does this condition check for? Can we add a comment or refactor the expression to add more clarity? This is because future readers of the code should be able to understand it.

Reply:
I will add a comment in the code as follows
// an overflow has happened on addition if the sign of the result is different from BOTH operands